### PR TITLE
Add a way to specify the state message

### PR DESCRIPTION
**Pull request by dbrgn**
The `state` object can contain a message. It could depend on the opening status.

### CODE REVIEW

**Comment by dbrgn** [comment]
@rnestler care to review? :)

File: spaceapi_server/src/server/handlers.rs [hunk 1]
```diff
@@ -65,14 +65,19 @@ impl ReadHandler {
...
+        let people_now_present: Option<u64> = status_copy.sensors.as_ref()
             .and_then(|sensors| sensors.people_now_present.as_ref())
-            .and_then(|people_now_present| {
-                match people_now_present[0].value {
-                    0i64 => Optional::Value(false),
-                    _ => Optional::Value(true),
-                }
-            }).into();
+            .map(|people_now_present| people_now_present[0].value)
+            .into();
```
Comment (rnestler) [inline spaceapi_server/src/server/handlers.rs]: Is the ``into()`` needed?

**Comment by rnestler** [comment]
So the messages are hardcoded into `spaceapi_server` now? I remember discussing the possibility to inject stuff like this from the main executable. But hardcoded is fine for now.

File: spaceapi_server/src/server/handlers.rs (hunk 1 above)
Comment (dbrgn) [inline spaceapi_server/src/server/handlers.rs]: Yes, it converts the Optional to an Option.

**Comment by dbrgn** [comment]
> So the messages are hardcoded into spaceapi_server now? I remember discussing the possibility to inject stuff like this from the main executable. But hardcoded is fine for now.

Yeah, that's #55, I'd say.

File: spaceapi_server/src/server/handlers.rs [hunk 2]
```diff
@@ -65,14 +65,19 @@ impl ReadHandler {
...
+        if let Some(count) = people_now_present {
+            status_copy.state.open = Some(count > 0);
+            if count == 1 {
+                status_copy.state.message = Optional::Value(format!("{} person here right now", count));
+            } else if count > 1 {
+                status_copy.state.message = Optional::Value(format!("{} people here right now", count));
+            }
```
Comment (rnestler) [inline spaceapi_server/src/server/handlers.rs]: What happens when `count == 0`? Then the value set when creating the server is used I guess. Now the messages are scattered around multiple places.

Comment (dbrgn) [inline spaceapi_server/src/server/handlers.rs]: Yes, I'm aware of that. But one of them is the static default message while the other one is dynamic depending on the people count. The dynamic part will be refactored out in #55.

Comment (rnestler) [inline spaceapi_server/src/server/handlers.rs]: Maybe something like
```Rust
    let people_now_present = Some(10u64);
    let open = people_now_present.map(|v|v>0);
    let message = match people_now_present {
        None|Some(0)=> "Open every Monday from 20:00".into(),
        Some(1) => format!("1 person here right now"),
        Some(count) => format!("{} people here right now", count),
    };    let people_now_present = Some(10u64);
    let open = people_now_present.map(|v|v>0);
    let message = match people_now_present {
        None|Some(0)=> "Open every Monday from 20:00".into(),
        Some(1) => format!("1 person here right now"),
        Some(count) => format!("{} people here right now", count),
    };
```
Would help readability.

Comment (dbrgn) [inline spaceapi_server/src/server/handlers.rs]: Hm, yes, now that I changed the type to u64 that would work :) So you'd skip the "default message" altogether?

File: spaceapi_server/src/server/handlers.rs (hunk 1 above)
Comment (rnestler) [inline spaceapi_server/src/server/handlers.rs]: Ah yes, forgot about `Optional`.

File: spaceapi_server/src/server/handlers.rs (hunk 2 above)
Comment (rnestler) [inline spaceapi_server/src/server/handlers.rs]: Yeah you're right. If it all gets injected by the user anyway in a later step there's no need to clean it up now.

**Pull request merged**